test/network_analysis/network_properties.jl

```diff
@@ -649,286 +320,103 @@ let
-    @test Catalyst.isdetailedbalanced(rn, rates) == true
-end
+### Tests for the matrices and vectors that are in the species-formation rate function
+# ẋ = S * K * Φ(t)
+# ẋ = Y * A_K * Φ(t)
 
-# Deficiency two network: the following reaction network must satisfy both the independent cycle conditions and the spanning forest conditions 
 let
-    rn = @reaction_network begin
-        (k1, k2), 3A <--> A + 2B
-        (k3, k4), A + 2B <--> 3B
-        (k5, k6), 3B <--> 2A + B
-        (k7, k8), 2A + B <--> 3A
-        (k9, k10), 3A <--> 3B
-    end
-
-    rcs, D = reactioncomplexes(rn)
-    @test Catalyst.edgeindex(D, 1, 2) == 1
-    @test Catalyst.edgeindex(D, 4, 3) == 6
-    k = rand(rng, numparams(rn))
-    p = parameters(rn)
-    rates = Dict(zip(parameters(rn), k))
-    @test Catalyst.isdetailedbalanced(rn, rates) == false
-
-    # Adjust rate constants to fulfill independent cycle conditions. 
-    rates[p[8]] = rates[p[7]]*rates[p[5]]*rates[p[9]] / (rates[p[6]]*rates[p[10]])
-    rates[p[3]] = rates[p[2]]*rates[p[4]]*rates[p[9]] / (rates[p[1]]*rates[p[10]])
-    @test Catalyst.isdetailedbalanced(rn, rates) == false
-    # Should still fail - doesn't satisfy spanning forest conditions. 
-
-    # Adjust rate constants to fulfill spanning forest conditions. 
-    cons = rates[p[6]] / rates[p[5]]
-    rates[p[1]] = rates[p[2]] * cons
-    rates[p[9]] = rates[p[10]] * cons^(3/2)
-    rates[p[8]] = rates[p[7]]*rates[p[5]]*rates[p[9]] / (rates[p[6]]*rates[p[10]])
-    rates[p[3]] = rates[p[2]]*rates[p[4]]*rates[p[9]] / (rates[p[1]]*rates[p[10]])
-    @test Catalyst.isdetailedbalanced(rn, rates) == true
-end
-
-### DEFICIENCY ONE TESTS
-
-# Fails because there are two terminal linkage classes in the linkage class
-let 
-    rn = @reaction_network begin
-        k1, A + B --> 2B
-        k2, A + B --> 2A
+    MAPK = @reaction_network MAPK begin
+        (k₁, k₂),KKK + E1 <--> KKKE1
+        k₃, KKKE1 --> KKK_ + E1
+        (k₄, k₅), KKK_ + E2 <--> KKKE2
+        k₆, KKKE2 --> KKK + E2
+        (k₇, k₈), KK + KKK_ <--> KK_KKK_
+        k₉, KK_KKK_ --> KKP + KKK_
+        (k₁₀, k₁₁), KKP + KKK_ <--> KKPKKK_
+        k₁₂, KKPKKK_ --> KKPP + KKK_
+        (k₁₃, k₁₄), KKP + KKPase <--> KKPKKPase
+        k₁₅, KKPPKKPase --> KKP + KKPase
+        k₁₆,KKPKKPase --> KK + KKPase
+        (k₁₇, k₁₈), KKPP + KKPase <--> KKPPKKPase
+        (k₁₉, k₂₀), KKPP + K <--> KKPPK
+        k₂₁, KKPPK --> KKPP + KP
+        (k₂₂, k₂₃), KKPP + KP <--> KPKKPP
+        k₂₄, KPKKPP --> KPP + KKPP
+        (k₂₅, k₂₆), KP + KPase <--> KPKPase
+        k₂₇, KKPPKPase --> KP + KPase
+        k₂₈, KPKPase --> K + KPase
+        (k₂₉, k₃₀), KPP + KPase <--> KKPPKPase
     end
 
-    @test Catalyst.satisfiesdeficiencyone(rn) == false
-end
+    Φ = Catalyst.massactionvector(MAPK)
+    specs = species(MAPK)
+    @test isequal(Φ[1], MAPK.KKK * MAPK.E1)
+    @test isequal(Φ[6], MAPK.KKK * MAPK.E2)
+    @test isequal(Φ[8], MAPK.KK_KKK_)
```

Comment:
Why not hand make the full vector and test the two vectors are equal (or `setequal`)?